control-core/src/actors/mitsubishi_inverter_rs485.rs

```diff
@@ -0,0 +1,171 @@
+    return (code & 0b10000000) > 0; // 0x80 is set when an exception happens
+}
+
+pub fn response_function_code_is_exception(function_code: u8) -> bool {
+    return (function_code & 0b10000000) > 0; // 0x80 is set when an exception happens
+}
+/*
+    Monitoring, operation command, frequency
+    setting (RAM)Less than 12 ms
+    Parameter read/write, frequency setting
+    (EEPROM)Less than 30 ms
+    Parameter clear / All parameter clearLess than 5 s
+    Reset commandNo reply
+*/
+enum RequestType {
+    OperationCommand,
+    ReadWrite,
+    ParamClear,
+    Reset,
+}
+
+impl RequestType {
+    fn timeout_milliseconds(self) -> u64 {
+        match self {
+            RequestType::OperationCommand => 12,
+            RequestType::ReadWrite => 30,
+            RequestType::ParamClear => 5000,
+            RequestType::Reset => 0,
+        }
+    }
+
+    fn timeout_nanoseconds(self) -> u64 {
+        self.timeout_milliseconds() * 1000000
+    }
+}
+
+enum MitsubishiModbusExceptionCode {
+    IllegalFunction,
+    IllegalDataAddress,
+    IllegalDataValue,
+    None,
+}
+
+impl MitsubishiModbusExceptionCode {
+    fn display(self) -> String {
+        match self {
+            MitsubishiModbusExceptionCode::IllegalFunction => "Illegal Function".to_string(),
+            MitsubishiModbusExceptionCode::IllegalDataAddress => "Illegal Data Address".to_string(),
+            MitsubishiModbusExceptionCode::IllegalDataValue => "Illegal Data Value".to_string(),
+            MitsubishiModbusExceptionCode::None => "No Exceptions".to_string(),
+        }
+    }
+}
+
+impl From<MitsubishiModbusExceptionCode> for u8 {
+    fn from(value: MitsubishiModbusExceptionCode) -> Self {
+        match value {
+            MitsubishiModbusExceptionCode::None => 0,
+            MitsubishiModbusExceptionCode::IllegalFunction => 1,
+            MitsubishiModbusExceptionCode::IllegalDataAddress => 2,
+            MitsubishiModbusExceptionCode::IllegalDataValue => 3,
+        }
+    }
+}
+
+impl From<u8> for MitsubishiModbusExceptionCode {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => MitsubishiModbusExceptionCode::IllegalFunction,
+            2 => MitsubishiModbusExceptionCode::IllegalDataAddress,
+            3 => MitsubishiModbusExceptionCode::IllegalDataValue,
+            _ => MitsubishiModbusExceptionCode::None,
+        }
+    }
+}
+
+impl Actor for MitsubishiInverterRS485Actor {
+    fn act(&mut self, _now_ts: Instant) -> Pin<Box<dyn Future<Output = ()> + Send + '_>> {
+        Box::pin(async move {
+            let elapsed: Duration = self.last_ts.elapsed();
+
+            // TODO: implement a message queue for the messages coming from machine api AND a response message queue maybe ?
+            let start_motor: Vec<u8> = ModbusRequest {
+                slave_id: 1,
+                function_code: ModbusFunctionCode::PresetHoldingRegister,
+                data: vec![0x0, 0x8, 00, 02],
```

Comment:
Make an enum with `MitsubishiInverterRS48Requests` which has a variant `StartMotor` (or something else), which  implements `From<MitsubishiInverterRS48Requests> for ModbusRequest`. The `vec!` also is a magic constant that needs to be explained.